### DISCUSSION TITLE
Did the new wagmi stop using batch for `useReadContract`?

### DISCUSSION THREAD

**Discussion by xsubject** [wevm/wagmi#3791] [Q&A]
I'm using wagmi 2.5.7

And I noticed that there are a lot of queries for some reason, I got interested in this and this is what I did, I set the default query to no refetching:
```typescript
export const UPDATE_POLICY_NO_UPDATE = {
  refetchInterval: false,
  staleTime: Infinity,
  refetchIntervalInBackground: false,
  refetchOnWindowFocus: false,
  refetchOnReconnect: false,
  refetchOnMount: false,
  retryOnMount: false
} as const

export const UPDATE_POLICY_FREQUENT = {
  refetchInterval: 5000,
  staleTime: 5000,
  refetchIntervalInBackground: false,
  refetchOnWindowFocus: true,
  refetchOnReconnect: true,
  refetchOnMount: false,
  retryOnMount: false
}

const queryClient = new QueryClient({
  defaultOptions: {
    queries: UPDATE_POLICY_NO_UPDATE,
  },
})
```

Now I've done something like:
```typescript
  const { data: balance } = useReadContract({
    abi: ...,
    address: ...,
    functionName: "balanceOf",
    args: [address as `0x${string}`],
    query: UPDATE_POLICY_FREQUENT
  })

  const { data: balance2 } = useReadContract({
    abi: ...,
    address: ...,
    functionName: "balanceOf",
    args: [address2 as `0x${string}`],
    query: UPDATE_POLICY_FREQUENT
  })
```

And I was surprised enough because he made 2 multicall calls(with single call). It seems from that moment on wagmi was no longer as good as it used to be. 
So it seems that now every single hook collects "HIS" batch. So if I use useReadContracts - then the batch will contain calls from this hook

I sincerely hope I'm missing something about the new wagmi :(

**Comment by tmm** [comment]
> new wagmi

what version were you using before `2.5.7`? nothing has changed with this behavior recently so a minimal reproduction is required to look into this anymore.
